recipes/libcurl/all/conanfile.py

```diff
@@ -285,9 +304,53 @@ def _configure_autotools_vars(self):
 
             del autotools_vars['LIBS']
             self.output.info("Autotools env vars: " + repr(autotools_vars))
+
+        if tools.cross_building(self.settings):
+            if self.settings.os == "iOS": 
+                ios_dev_target = str(self.settings.os.version).split(".")[0]
+                if self.settings.arch in ["x86", "x86_64"]:
+                    iphoneos = "iphonesimulator"
+                    autotools_vars['CPPFLAGS'] = "-D__IPHONE_OS_VERSION_MIN_REQUIRED={}0000".format(ios_dev_target)
+                elif self.settings.arch in ["armv7", "armv7s", "armv8"]:
+                    iphoneos = "iphoneos"
+                    autotools_vars['CPPFLAGS'] = ""
+                else:
+                    raise ConanInvalidConfiguration("Unsuported iOS arch {}".format(self.settings.arch)) 
+
+                cc = tools.XCRun(self.settings, iphoneos).cc
+                sysroot = "-isysroot {}".format(tools.XCRun(self.settings, iphoneos).sdk_path)
+
+                if self.settings.arch == "armv8":
+                    configure_arch = "arm64"    
+                    configure_host = "arm" #unused, autodetected
+                else:
+                    configure_arch = self.settings.arch
+                    configure_host = self.settings.arch #unused, autodetected
+    
+
+                arch_flag = "-arch {}".format(configure_arch)
+                ios_min_version = "-miphoneos-version-min={}".format(ios_dev_target)
+                bitcode =  "-fembed-bitcode" if self.options.enable_bitcode else ""
+                extra_flag = "-Werror=partial-availability"
+                extra_def = " -DHAVE_SOCKET -DHAVE_FCNTL_O_NONBLOCK"
+                # if we debug, maybe add a -gdwarf-2 , but why would we want that?  
+
+                autotools_vars['CC'] = cc 
+                autotools_vars['IPHONEOS_DEPLOYMENT_TARGET'] = ios_dev_target
+                autotools_vars['CFLAGS'] = "{} {} {} {} {}".format(
+                    sysroot, arch_flag, ios_min_version, bitcode, extra_flag
+                ) 
+                autotools_vars['LDFLAGS'] = "{} {}".format(arch_flag, sysroot)
```

Comment:
I guess it would be better to manipulate the properties of `AutoToolsBuildEnvironment` (cflags, link_flags) instead of overriding environment variables. the primary reason that user may have set `CFLAGS`/`LDFLAGS` within his conan profile, and such flags would be lost here.

Reply:
will check what is possible, thanks!

Reply:
does not work, with not using the environment variables 

reduced to the smallest example, using the `-isysroot` flag

having it in the environment
```
configure:4717: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang  -c -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk -arch armv7 -miphoneos-version-min=10  -Werror=partial-availability  -DHAVE_SOCKET -DHAVE_FCNTL_O_NONBLOCK conftest.c >&5
configure:4717: $? = 0
configure:4730: result: none needed
```

using `self._autotools.flags.append(sysroot)` , where sysroot is the exact same path
```
configure:4717: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang  -c -arch armv7 -miphoneos-version-min=10  -Werror=partial-availability  -DHAVE_SOCKET -DHAVE_FCNTL_O_NONBLOCK conftest.c >&5
conftest.c:10:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
configure:4717: $? = 1
configure: failed program was:
......
```

so obviously, this is not the same for configure, the values are not passed on

I have now invested some hours in this and will not continue not using the env_var, if you think you can improve, please provide a working example